src/main/java/com/microsoft/azure/AzureVMManagementServiceDelegate.java

```diff
@@ -632,14 +638,16 @@ public static String verifyConfiguration(
             return Messages.Azure_GC_Template_Val_Profile_Missing();
         } else {
             try {
+                if (!resourceGroupName.matches(Constants.DEFAULT_RESOURCE_GROUP_PATTERN)) {
```

Comment:
serviceprincipal.validate  checkes arg, vm limit and timeout, here we only check rg. If we need to check all 3 then we need to change verify configuration as well. should I do it?